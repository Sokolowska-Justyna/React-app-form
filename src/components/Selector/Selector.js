import React from "react";
import "./Selector.scss";

export function Selector(props) {
  const { currentValue, values, emptyValueText, onChange, name } = props;
  return (
      <select
        name={name}
        value={currentValue}
        required
        onChange={onChange}
        className="select-input"
      >
        <option key="" value="" disabled>
          {emptyValueText}
        </option>
        
        {values.map(val => {
          const key = Object.keys(val);
          return (
            <option key={key} value={key}>
              {val[key]}
            </option>
          );
        })}
      </select>
  );
}
