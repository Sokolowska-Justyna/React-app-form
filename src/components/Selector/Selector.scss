.select-input {
    border-radius: 4px;
    font-size: 12px;
    border: 1px solid #cccccc;
    margin-top: 13px;
    width: 100%;
    height: 34px;
    background-color: #ffffff;
    color: #434a54;
    padding-left: 19px;
    font-weight: bold;
    @media only screen and (max-width: 430px) {
        margin-top:5px;
    }
}