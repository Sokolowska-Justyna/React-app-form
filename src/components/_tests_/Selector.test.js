import React from 'react';
import { Selector } from '../Selector';
import { shallow } from 'enzyme';

describe('Selector', () => {
  it('should match snapshot', () => {
    const options = ['option1', 'option2'];
    const wrapper = shallow(<Selector 
      name="selector-name"
      currentValue={'option1'}
      values={options}
      onChange={jest.fn()}
    />);
  
    expect(wrapper).toMatchSnapshot();
  });
});