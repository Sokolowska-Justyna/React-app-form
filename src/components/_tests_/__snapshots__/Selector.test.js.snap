// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Selector
    currentValue="option1"
    name="selector-name"
    onChange={[MockFunction]}
    values={
      Array [
        "option1",
        "option2",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          disabled={true}
          value=""
        />,
        Array [
          <option
            value={
              Array [
                "0",
                "1",
                "2",
                "3",
                "4",
                "5",
                "6",
              ]
            }
          />,
          <option
            value={
              Array [
                "0",
                "1",
                "2",
                "3",
                "4",
                "5",
                "6",
              ]
            }
          />,
        ],
      ],
      "name": "selector-name",
      "onChange": [MockFunction],
      "required": true,
      "value": "option1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "",
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "disabled": true,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "0,1,2,3,4,5,6",
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "value": Array [
            "0",
            "1",
            "2",
            "3",
            "4",
            "5",
            "6",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "0,1,2,3,4,5,6",
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "value": Array [
            "0",
            "1",
            "2",
            "3",
            "4",
            "5",
            "6",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            disabled={true}
            value=""
          />,
          Array [
            <option
              value={
                Array [
                  "0",
                  "1",
                  "2",
                  "3",
                  "4",
                  "5",
                  "6",
                ]
              }
            />,
            <option
              value={
                Array [
                  "0",
                  "1",
                  "2",
                  "3",
                  "4",
                  "5",
                  "6",
                ]
              }
            />,
          ],
        ],
        "name": "selector-name",
        "onChange": [MockFunction],
        "required": true,
        "value": "option1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "disabled": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "0,1,2,3,4,5,6",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "value": Array [
              "0",
              "1",
              "2",
              "3",
              "4",
              "5",
              "6",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "0,1,2,3,4,5,6",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "value": Array [
              "0",
              "1",
              "2",
              "3",
              "4",
              "5",
              "6",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
